python/ray/tests/test_actor_failures.py

```diff
@@ -891,6 +892,141 @@ def ping(self):
     assert wait_for_condition(lambda: ray.get(parent_actor.wait.remote()))
 
 
+@pytest.mark.parametrize(
+    "ray_start_cluster", [{
+        "num_cpus": 1,
+        "num_nodes": 1,
+    }], indirect=True)
+def test_location_resolution_worker_failure(ray_start_cluster):
+    """Test location resolution protocol when actor is not persisted to GCS
+    This test verifies the scneario where owner worker
+    has failed before actor is persisted to GCS.
+    Reference: https://github.com/ray-project/ray/pull/8679
+    """
+
+    @ray.remote
+    class Actor:
+        def __init__(self, dependency):
+            print("actor: {}".format(os.getpid()))
+            self.dependency = dependency
+
+        def f(self):
+            return self.dependency
+
+    @ray.remote
+    class Owner:
+        def get_pid(self):
+            return os.getpid()
+
+        def create_actor(self, caller_handle):
+            s = SignalActor.remote()
```

Comment:
Can we simplify use a long-running remote task as a dependency here?
Not sure why need this `SignalActor`, this makes the code hard to understand.

Reply:
We've been using `SignalActor` in lots of other unit tests to avoid having to rely on timing. So I think it actually makes perfect sense to use here, but I don't feel strongly about it.